### ISSUE TITLE
mismatching issue with when path contains '[***]

### ISSUE THREAD

**Issue by qvandekadsye** [isaacs/node-glob#277]
Hi, I encounter some issue with matching using glob : 

I'm a currently using this pattern to find my mp3 file "path.join(currentdirPath,"/*.mp3") where currentdir change thanks to a parameters.

I noticed when currentdirpath contains "[somecharacter]" like in this exemple : 
Senki Zesshou Symphogear Original Soundtrack 1 [mp3], glob found nothing.
but in this exemple :  Senki Zesshou Symphogear Character Song 1 - Zwei Wing

glob found all the mp3 files

i think that's beacause [...] is a pattern, but how can I escape [ and ]

**Comment by hhamilto** [comment]
>how can I escape [ and ]

How about with back slashes like this:

    currentdirPath.replace(/[[\]]/g,'\\$&')

That replaces `[` with `\[` and `]` with `\]`, which causes them to be treated as literal opening and closing bracket characters.

I imagine that there are other edge cases that should be taken care of when your path might contain special characters, but I'm not sure about how to handle all cases off the top of my head.

**Comment by hhamilto** [comment]
I was looking through the documentation just now and found a better solution to your problem, rather than escaping `currentdirPath`. 

You should pass in the `cwd` property as `currentdirPath` on the options object when calling glob. eg:

    glob('*.mp3', {cwd: currentdirPath})

**Comment by sinedied** [comment]
@hhamilto I have a similar issue, except that the [ and ] character are part of the filename. I tried escaping with a backslash using your regexp, but it still doesn't work. Any ideas?

**Comment by woojoo666** [comment]
confirmed as of today (version 7.1.3) that escaping square brackets (eg `\\[`) does not work. I have a suspicion it's because it supports windows-style paths, eg this is a path that glob supports: `C:\\this\\is\\a\\windows\\path`. However, if you try to escape brackets, eg `C:\\path\\[hi\\]`, then how does it differentiate between interpretting it as a path separator (equivalent to "C:/path/[hi/]") or as an escape character (equivalent to "C:/path/[hi]"

**Comment by woojoo666** [comment]
actually, as I was typing the previous comment, I came up with a devious hack: just wrap all square brackets with `[...]`, so it gets treated as a character in the match pattern.

Eg if you want to match the path `C:/foo [bar]` use the match pattern `C:/foo [[]bar[]]`

Ugly, but works, even as far back as package version 7.1.1 (that's as far back as I checked). This should be put in the ReadMe though because it's the only solution I've found so far.
